### DISCUSSION TITLE
Idea: Different overlay effects

### DISCUSSION THREAD

**Discussion by svenjacobs** [svenjacobs/reveal#44] [Ideas]
Currently we only have one overlay effect, which is darkening the background. We could implement different overlay composables with new effects like blurring or pixelating the background, if this is natively possible with Compose.

**Comment by rajkumaranbu94** [comment]
Hi @svenjacobs,

Would you consider adding support for below features to the library?

- Currently, the overlay only darkens the background. It would be awesome to support different effects like light transparent overlays or blur effects behind the highlighted element. Since Jetpack Compose supports blur (via RenderEffect on Android 12+), this could be a powerful enhancement.It would allow developers to easily implement different visual effects like blurring, pixelation, or lighter overlays.
- And sometimes the default padding around the revealed element feels too large. It would be great to have an option to disable or customize this padding so that only the arrowed bubble coachmark is visible, making the highlight cleaner and less intrusive.

They would really help tailor coachmarks to various UI styles.

**Reply by svenjacobs** [reply]
Hello @rajkumaranbu94,

> Currently, the overlay only darkens the background. It would be awesome to support different effects like light transparent overlays or blur effects behind the highlighted element. Since Jetpack Compose supports blur (via RenderEffect on Android 12+), this could be a powerful enhancement.It would allow developers to easily implement different visual effects like blurring, pixelation, or lighter overlays.

Yes, the foundation is already there for different effects. Since Reveal is a multiplatform library it would be nice if the effects work on all platforms. However I'm not so sure about blurring.

Is this something that you might want to investigate and provide a PR?

> And sometimes the default padding around the revealed element feels too large. It would be great to have an option to disable or customize this padding so that only the arrowed bubble coachmark is visible, making the highlight cleaner and less intrusive.

The `revealable` modifier already has a `padding` argument. Is that what you're looking for?

**Reply by rajkumaranbu94** [reply]
Hello @svenjacobs,

First off, thank you for the great work on the Reveal library — it's really close to what I need for a smooth coach mark experience in my app.

**My requirement is fairly straightforward:**

I'd like a light transparent overlay instead of the default dark one.

I’d prefer not to highlight the element with padding or a shape — just showing a bubble with an arrow pointing to the element would be ideal.

If there's a way to support this natively, I’d be happy to contribute to the library to make it happen.

Thanks again for your efforts!

**Reply by svenjacobs** [reply]
> I'd like a light transparent overlay instead of the default dark one.

A background effect that does nothing should be fairly simple 😉 Since the reveal area around a revealable item is subtracted (removed) from the background effect, there would be no reveal area with a transparent background. I think that is what you're looking for? Do you want to investigate and create a PR?

**Reply by svenjacobs** [reply]
Another solution just came to my mind: Have you tried setting the background color of the dim effect to a transparent color?

**Reply by rajkumaranbu94** [reply]
> Another solution just came to my mind: Have you tried setting the background color of the dim effect to a transparent color?

Hi @svenjacobs , I tried but its applied on top the existing background color.

**Reply by rajkumaranbu94** [reply]
> > I'd like a light transparent overlay instead of the default dark one.
> 
> A background effect that does nothing should be fairly simple 😉 Since the reveal area around a revealable item is subtracted (removed) from the background effect, there would be no reveal area with a transparent background. I think that is what you're looking for? Do you want to investigate and create a PR?

Hi @svenjacobs , Yes right, I created my own custom one as per my requirement using native compose for android. If its helps you, then I can investigate and provide you PR.

**Reply by svenjacobs** [reply]
> Hi @svenjacobs , I tried but its applied on top the existing background color.

I tried this in the [demo app](https://github.com/svenjacobs/reveal/tree/main/demo-app) and it seems to work? I added

```
overlayEffect = DimRevealOverlayEffect(color = Color.Transparent),
```

to `Reveal` and removed the `borderStroke` from the `revealable` modifiers and this is what I get:

<img width="640" height="1428" alt="Screenshot_20250724_123420" src="https://github.com/user-attachments/assets/2bcafa3f-7159-441e-bee8-9164748d80d9" />

Isn't that what you are looking for? Please provide a screenshot when it's not working for you.

**Reply by rajkumaranbu94** [reply]
Hi @svenjacobs, This is cool, I tried its working for me too. Thank you so much!
Then the only issue is, showing pointer arrow always on centre means if target element is on the edge then bubble will be hided on off screen. Here the screenshot for your reference,
<img width="1080" height="2408" alt="revealsc" src="https://github.com/user-attachments/assets/2d587d26-ec97-45d3-97c6-de1029aeb39c" />
